frontend/packages/dev-console/src/utils/__tests__/pipeline-augment.spec.ts

```diff
@@ -328,3 +296,38 @@ describe('Pipeline exists test to determine whether a pipeline is linked to a pi
     expect(pipelineFound).toBeFalsy();
   });
 });
+
+describe('Pipelinerun graph to show the executed pipeline structure', () => {
+  const currentPipeline = pipelineTestData[PipelineExampleNames.SIMPLE_PIPELINE].pipeline;
+  const testPipelineRun =
+    pipelineTestData[PipelineExampleNames.SIMPLE_PIPELINE].pipelineRuns[DataState.SUCCESS];
+
+  it('expect to return null, if pipelinerun does not have a status field', () => {
+    const plrWithoutPipelineSpec = _.omit(testPipelineRun, ['status']);
+    const executedPipeline = getPipelineFromPipelineRun(plrWithoutPipelineSpec);
+    expect(executedPipeline).toEqual(null);
+  });
+
+  it('expect to return null, if pipelinerun does not have pipelineSpec in status field', () => {
+    const executedPipeline = getPipelineFromPipelineRun(
+      _.omit(testPipelineRun, ['status.pipelineSpec']),
+    );
+    expect(executedPipeline).toBe(null);
+  });
+
+  it('expect to return the previous pipeline structure, if pipelinerun has pipelineSpec in status field', () => {
+    const plrWithOneTask = {
+      ...testPipelineRun,
+      status: {
+        pipelineSpec: {
+          ...currentPipeline.spec,
+          tasks: [currentPipeline.spec.tasks[0]],
+        },
+      },
+    };
+    const executedPipeline = getPipelineFromPipelineRun(plrWithOneTask);
+    expect(currentPipeline.spec.tasks).toHaveLength(2);
+    expect(executedPipeline.spec.tasks).toHaveLength(1);
+    expect(executedPipeline).not.toEqual(currentPipeline);
```

Comment:
Confused about this too... this follows my previous comment's confusion.

Reply:
Updated the test